### PR TITLE
doc/user-guide.md: add docs for cli cmd build

### PR DESCRIPTION
**Pull request by srleyva**
started documentation for operator-sdk and added docs for build cmd

Refers #266

Is this a good place for the CLI documentation? @fanminshi

### CODE REVIEW

File: doc/sdk-cli-reference.md [hunk 1]
```diff
@@ -0,0 +1,169 @@
...
+cd $GOPATH/src/github.com/example.com/
+operator-sdk new app-operator --api-version=app.example.com/v1alpha1 --kind=AppService
+
+//Output
+Create app-operator/.gitignore
+...
+
+```
+
+## up
```
Comment (fanminshi) [inline doc/sdk-cli-reference.md]: The up command was recently updated with more feature. could you update the this doc along with those?

**Comment by fanminshi** [comment]
@srleyva sorry for the delay. I am waiting for a fix https://github.com/operator-framework/operator-sdk/pull/274 and https://github.com/operator-framework/operator-sdk/pull/264 before commenting on this doc.

The doc generally looks good.

**Comment by fanminshi** [comment]
cc/ @rithujohn191

File: doc/sdk-cli-reference.md (hunk 1 above)
Comment (srleyva) [reply doc/sdk-cli-reference.md]: Added the new features to the doc

File: doc/sdk-cli-reference.md [hunk 2]
```diff
@@ -0,0 +1,177 @@
...
+
+##### Use
+
+k8s generator generates code for custom resource given the API spec
+to comply with kube-API requirements.
+
+##### Flags
+
+* `-h, --help` - help for k8s
+
```
Comment (etiennecoutaud) [inline doc/sdk-cli-reference.md]: Maybe we could add an `Example` section to be consistent

File: doc/sdk-cli-reference.md [hunk 3]
```diff
@@ -0,0 +1,177 @@
...
+* `--kubeconfig` string - The file path to kubernetes configuration file; defaults to $HOME/.kube/config
+
+* `--namespace` string - The namespace where the operator watches for changes. (default "default")
+
+* `--operator-flags` - Flags that the local operator may need.
+
+* `-h, --help` - help for local
+
+##### Example
```
Comment (fanminshi) [inline doc/sdk-cli-reference.md]: I'll let @shawn-hurley to provide additional descriptions on how to use those flags as he worked on this in a future PR.

File: doc/sdk-cli-reference.md (hunk 2 above)
Comment (srleyva) [reply doc/sdk-cli-reference.md]: Added it...also tried to keep the consistency in the generate olm-catalog doc as well.
Thanks for the suggestion!

File: doc/sdk-cli-reference.md [hunk 4]
```diff
@@ -0,0 +1,193 @@
...
+##### Flags
+
+* `-h, --help` - help for k8s
+
+##### Example
+
+```bash
+operator-sdk generate k8s
+
+# Output
```
Comment (fanminshi) [inline doc/sdk-cli-reference.md]: missed a `:`

File: doc/sdk-cli-reference.md [hunk 5]
```diff
@@ -0,0 +1,193 @@
...
+* `--image` **(required)** string - The container image name to set in the CSV to deploy the operator e.g: quay.io/example/operator:v0.0.1
+* `--version` **(required)** string - The version of the current CSV e.g: 0.0.1
+* `-h, --help` - help for olm-catalog
+
+##### Example
+
+```bash
+operator-sdk generate olm-catalog --image=quay.io/example/operator:v0.0.1 --version=0.0.1
+
+# Output
```
Comment (fanminshi) [inline doc/sdk-cli-reference.md]: missed `:`

File: doc/sdk-cli-reference.md [hunk 6]
```diff
@@ -0,0 +1,193 @@
...
+* `-h, --help` - help for new
+
+### Example
+
+```bash
+mkdir $GOPATH/src/github.com/example.com/
+cd $GOPATH/src/github.com/example.com/
+operator-sdk new app-operator --api-version=app.example.com/v1alpha1 --kind=AppService
+
+//Output
```
Comment (fanminshi) [inline doc/sdk-cli-reference.md]: `//Output` -> `# Output:`?

File: doc/sdk-cli-reference.md [hunk 7]
```diff
@@ -0,0 +1,193 @@
...
+Create app-operator/.gitignore
+...
+
```
Comment (fanminshi) [inline doc/sdk-cli-reference.md]: remove the extra line?

File: doc/sdk-cli-reference.md [hunk 8]
```diff
@@ -0,0 +1,193 @@
...
+##### Example
+
+```bash
+operator-sdk generate k8s
+
+# Output
+Run code-generation for custom resources
+Generating deepcopy funcs
+
```
Comment (fanminshi) [inline doc/sdk-cli-reference.md]: No need to have the extra line here.

**Comment by fanminshi** [comment]
lgtm after nits.

**Comment by srleyva** [comment]
Thanks for the help @fanminshi

**Pull request merged**